src/react-imgix.jsx

```diff
@@ -109,68 +110,97 @@ function buildSrc({
+  if (fixedSize) {
+    const { q, ...urlParams } = srcOptions;
+    const params = {};
 
-  if (disableSrcSet) {
-    srcSet = src;
-  } else {
-    if (fixedSize) {
-      const { q, ...urlParams } = srcOptions;
-      const constructedUrl = constructUrl(rawSrc, urlParams);
-
-      let srcFn = buildDprSrcWithQualityByDpr;
-      if (q) {
-        srcFn = buildDprSrcWithQuality;
-      } else if (disableQualityByDPR) {
-        srcFn = buildDprSrcWithoutQuality;
+    for (const k in urlParams) {
+      if (PARAM_EXPANSION[k]) {
+        params[PARAM_EXPANSION[k]] = urlParams[k];
+      } else {
+        params[k] = urlParams[k];
       }
+    }
 
-      srcSet = "";
-      const len = DPR_QUALITY_VALUES.length;
-      for (let i = 0; i < len; i++) {
-        const quality = DPR_QUALITY_VALUES[i];
-        srcSet += srcFn(constructedUrl, q || quality, i + 1) + ", ";
-      }
-      srcSet = srcSet.slice(0, -2);
-    } else {
-      const { width, w, height, ...urlParams } = srcOptions;
-      const constructedUrl = constructUrl(rawSrc, urlParams);
+    if (width) {
+      params["w"] = width;
+    }
 
-      const aspectRatio = imgixParams.ar;
-      let showARWarning =
-        aspectRatio != null && aspectRatioIsValid(aspectRatio) === false;
+    if (height) {
+      params["h"] = height;
+    }
 
-      let srcFn = buildSrcSetPairWithTargetWidth;
-      if (height) {
-        srcFn = buildSrcSetPairWithFixedHeight;
-      }
+    const [scheme, rest] = rawSrc.split("://");
+    const [domain, ...pathComponents] = rest.split("/");
+    let useHTTPS = scheme == "https";
+
+    const client = new ImgixClient({
+      domain: domain,
+      useHTTPS: useHTTPS,
+      includeLibraryParam: false,
+    });
 
-      srcSet = "";
-      const len = targetWidths.length;
-      for (let i = 0; i < len; i++) {
-        const targetWidth = targetWidths[i];
-        srcSet += srcFn(constructedUrl, targetWidth, height) + ", ";
+    const srcSet = client.buildSrcSet(
+      pathComponents.join("/"),
+      { ...params, q },
+      {
+        disableVariableQuality: disableQualityByDPR,
       }
-      srcSet = srcSet.slice(0, -2);
+    );
 
-      if (
-        NODE_ENV !== "production" &&
-        showARWarning &&
-        config.warnings.invalidARFormat
-      ) {
-        console.warn(
-          `[Imgix] The aspect ratio passed ("${aspectRatio}") is not in the correct format. The correct format is "W:H".`
-        );
+    const src = constructUrl(rawSrc, srcOptions);
+    return { src, srcSet };
+  } else {
+    const { width, w, height, h, ...urlParams } = srcOptions;
+    const params = {};
+
+    for (const k in urlParams) {
+      if (PARAM_EXPANSION[k]) {
+        params[PARAM_EXPANSION[k]] = urlParams[k];
+      } else {
+        params[k] = urlParams[k];
       }
     }
-  }
 
-  return {
-    src,
-    srcSet,
-  };
+    if (width) {
+      params["w"] = width ? width : w;
+    }
+
+    if (height) {
+      params["h"] = height || h;
+    }
```

Comment:
Nit. Why are we assigning conditionally values differently here? Ie `bar = foo ? foo : f`  vs `bar = foo || f`. Both `width` and `height` are declared in this closure no? So we can just use `bar = foo || f` for both w and h.

Reply:
Ooh, yeah. I had experimented with something else here and left it inconsistent. I can change this.